hasktorch/src/Torch/Typed/Native.hs

```diff
@@ -1987,8 +1987,22 @@ det input = unsafePerformIO $ ATen.cast1 ATen.Managed.det_t input
 -- diagonal :: Tensor device dtype shape -> Int -> Int -> Int -> Tensor device dtype shape
 -- diagonal _input _offset _dim1 _dim2 = unsafePerformIO $ (ATen.cast4 ATen.Managed.diagonal_tlll) _input _offset _dim1 _dim2
 
--- dot :: Tensor device dtype shape -> Tensor device dtype shape -> Tensor device dtype shape
--- dot _input _tensor = unsafePerformIO $ (ATen.cast2 ATen.Managed.dot_tt) _input _tensor
+type family DotDTypeIsValid (device :: (D.DeviceType, Nat)) (dtype :: D.DType) :: Constraint where
+  DotDTypeIsValid '( 'D.CPU, 0)            dtype = ( DTypeIsNotBool '( 'D.CPU, 0) dtype
+                                                   , DTypeIsNotHalf '( 'D.CPU, 0) dtype
+                                                   )
+  DotDTypeIsValid '( 'D.CUDA, deviceIndex) dtype = DTypeIsNotBool '( 'D.CUDA, deviceIndex) dtype
```

Comment:
this is NOT correct and needs to be further constrained to floating point types!